lib/matplotlib/_mathtext.py

```diff
@@ -1817,8 +1817,11 @@ def __init__(self):
         def set_names_and_parse_actions():
             for key, val in vars(p).items():
                 if not key.startswith('_'):
-                    # Set names on everything -- very useful for debugging
-                    val.setName(key)
+                    # Set names on (almost) everything -- very useful for debugging
+                    # token, placeable, and auto_delim are forward references which
+                    # are left without names to ensure useful error messages
+                    if key not in ("token", "placeable", "auto_delim"):
```

Comment:
Impact should be pretty minimal for earlier versions of pyparsing. Essentially if one of these raises an error itself it gets more verbose... but I'm having trouble even making these error (themselves and not a deeper constituent piece) when I try to, so not totally convinced those error message _ever_ happen.

On pyparsing 3.1.0, these error messages override the constituent parts, and thus are more visible, so it goes from

```
Expected token, got ...
```

To:

```
Expected Forward {{ simple | ... } | unknown_symbol}, got ...
```

Which could be better or worse depending on context, but the idea is to facilitate better errors in as yet unreleased pyparsing:

And on future versions of pyparsing, assuming we complete the line drawing we have been discussing, this will signal to return to pre 3.1.0 behavior (at least with many of the proposals).


Personally, I think moving the tuple negatively impacts readablity for rather minimal performance gains (this function is only called at parser init time, though admittedly multiple times, and the for loop has relatively few entries (~13 in the first call, ~40 in the second call, prior to this PR it is called 3 times, but I found no need for it to be called both before and after adding the `Forward` definitions, which don't rely on it having been called prior)) So moving would be saving ~50 tuple definitions, total per python process (even the parser object gets cached as a class variable and reused via the public API, so even creating a new `MathTextParser` does not rerun this). And even then, not convinced the variable lookup would be any faster

I think we are talking ~1 microsecond, total.

Reply:
I always have a hard time judging minor impact on readability and minor impact on performance and will always go for performance...